airbyte-integrations/connectors/source-chartmogul/unit_tests/test_source.py

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from unittest.mock import ANY, MagicMock
+
+from source_chartmogul.source import SourceChartmogul
+
+
+def test_check_connection(mocker, requests_mock):
+    source = SourceChartmogul()
+    logger_mock, config_mock = MagicMock(), MagicMock()
+
+    # success
+    requests_mock.get("https://api.chartmogul.com/v1/ping", json={"data": "pong!"})
+    assert source.check_connection(logger_mock, config_mock) == (True, None)
+
+    # failure
+    requests_mock.get("https://api.chartmogul.com/v1/ping", status_code=500)
+    assert source.check_connection(logger_mock, config_mock) == (False, ANY)
```

Comment:
I think the error is of type `HTTPError` if you used `raise_for_status`, you could test the error is of this type.